website/src/pages/docs/guides/graphql-server-apollo-yoga-with-server-preset.mdx

```diff
@@ -0,0 +1,349 @@
+
+```text
+├── src/
+│   ├── schema/
+│   │   ├── base/
+│   │   │   ├── schema.graphql
+│   │   ├── user/
+│   │   │   ├── schema.graphql
+│   │   ├── book/
+│   │   │   ├── schema.graphql
+```
+
+Here's the content of each module schema:
+
+```graphql
+# src/schema/base.graphql
+type Query
+type Mutation
+
+# src/schema/user.graphql
+extend type Query {
+  user(id: ID!): User
+}
+type User {
+  id: ID!
+  fullName: String!
+}
+
+# src/schema/book.graphql
+extend type Query {
+  book(id: ID!): Book
+}
+extend type Mutation {
+  markBookAsRead(id: ID!): Book!
+}
+type Book {
+  id: ID!
+  isbn: String!
+}
+```
+
+#### 2. Install Server Preset
+
+<PackageCmd packages={['-D @eddeee888/gcg-typescript-resolver-files']} />
+
+#### 3. Configure Codegen Config
+
+Create or update your `codegen.ts` as follows:
+
+```ts filename="codegen.ts"
+import type { CodegenConfig } from '@graphql-codegen/cli'
+import { preset } from '@eddeee888/gcg-typescript-resolver-files'
+
+const config: CodegenConfig = {
+  schema: '**/schema.graphql',
+  generates: {
+    'src/schema': {
+      preset,
+      plugins: []
+    }
+  }
+}
+export default config
+```
+
+<Callout type="info">
+
+**Remove existing `@graphql-codegen/typescript` and `@graphql-codegen/typescript-resolvers` config**
+
+The server preset comes with these plugins built-in with better defaults. You can override the defaults using [typesPluginsConfig option](https://github.com/eddeee888/graphql-code-generator-plugins/tree/master/packages/typescript-resolver-files#config).
+
+</Callout>
+
+### Generate Files
+
+Now, run codegen:
+
+<PackageCmd packages={[{ name: 'graphql-codegen', cmd: 'run', isNpx: true }]} />
+
+This results in the following structure:
+
+```
+├── src/
+│   ├── schema/
+│   │   ├── base/
+│   │   │   ├── schema.graphql
+│   │   ├── user/
+│   │   │   ├── resolvers/
+│   │   │   │   ├── Query/
+│   │   │   │   │   ├── user.ts            # Generated, changes not overwritten by codegen
+│   │   │   │   ├── User.ts                # Generated, changes not overwritten by codegen
+│   │   │   ├── schema.graphql
+│   │   ├── book/
+│   │   │   ├── resolvers/
+│   │   │   │   ├── Query/
+│   │   │   │   │   ├── book.ts            # Generated, changes not overwritten by codegen
+│   │   │   │   ├── Mutation/
+│   │   │   │   │   ├── markBookAsRead.ts  # Generated, changes not overwritten by codegen
+│   │   │   │   ├── Book.ts                # Generated, changes not overwritten by codegen
+│   │   │   ├── schema.graphql
+│   │   ├── resolvers.generated.ts         # Entirely generated by codegen
+│   │   ├── typesDefs.generated.ts         # Entirely generated by codegen
+│   │   ├── types.generated.ts             # Entirely generated by codegen
+```
+
+#### Generated Files Overview
+
+- `types.generated.ts`: TypeScript types generated by `@graphql-codegen/typescript` and `@graphql-codegen/typescript-resolvers`
+- `typeDefs.generated.ts`: Static TypeScript Schema AST to be used by the server
```

Comment:
This feature is not relased yet: https://github.com/eddeee888/graphql-code-generator-plugins/pull/51

I'll do some final testings today

Reply:
This is done